src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -1,78 +1,154 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-
 using System;
 using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Net.Http;
 using System.IO;
-using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
-using System.Net.Http.Headers;
+using Microsoft.Common.Core;
 
 namespace Microsoft.R.Host.Broker.RemoteUri {
     public class RemoteUriHelper {
         public static async Task HandlerAsync(HttpContext context) {
-            BinaryReader reader = new BinaryReader(context.Request.Body);
-            var url = reader.ReadString();
-            var method = reader.ReadString();
-            Dictionary<string, string> headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadString());
+            var url = context.Request.Headers["x-rtvs-url"];
```

Comment:
Since it's used on both sides, let's put it as a named constant somewhere in the `Microsoft.R.Host.Protocol` assembly.

Also, I believe the naming convention for these would make it something like "X-RTVS-URL". Maybe worth clarifying, too - something like "X-RTVS-Requested-URL".